CorsixTH/Lua/HeatingController.lua

```diff
@@ -0,0 +1,182 @@
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+class "HeatingController"
+
+---@type HeatingController
+local HeatingController = _G["HeatingController"]
+
+function HeatingController:HeatingController(world)
+  self.world = world
+  self.radiator_heat = 0.5
+  self:resetBoilerBreakdownCounter()
+  self.acc_heating = 0
+end
+
+function HeatingController:resetBoilerBreakdownCounter()
+  self.days_until_boiler_breakdown = self.world.map.level_config.gbv.DisasterLaunch
+  --if disasterLaunch not defined in Levelfile
+  if self.days_until_boiler_breakdown == nil then
+    self.days_until_boiler_breakdown = 200
+  end
+end
+
+function HeatingController:checkHeatingFacilities(day,averagePatiensWarmth, averageStaffWarmth)
+  if not self.warmth_msg and not self.heating_broke then
+    if day == 15 then
+      if averagePatiensWarmth < 0.22 then
+        self:warningTooCold()
+      elseif averagePatiensWarmth >= 0.36 then
+        self:warningTooHot()
+      end
+    end
+    -- Are the staff warm enough?
+    if day == 20 then
+      if averageStaffWarmth < 0.22 then
+        self.world.ui.adviser:say(_A.warnings.staff_very_cold)
+      elseif averageStaffWarmth >= 0.36 then
+        self.world.ui.adviser:say(_A.warnings.staff_too_hot)
+      end
+    end
+  end
+end
+
+function HeatingController:warningTooCold()
+  local cold_msg = {
+    (_A.information.initial_general_advice.increase_heating),
+    (_A.warnings.patients_very_cold),
+    (_A.warnings.people_freezing),
+  }
+  if cold_msg then
+    self.world.ui.adviser:say(cold_msg[math.random(1, #cold_msg)])
+    self.warmth_msg = true
+  end
+end
+
+function HeatingController:warningTooHot()
+  local hot_msg = {
+    (_A.information.initial_general_advice.decrease_heating),
+    (_A.warnings.patients_too_hot),
+    (_A.warnings.patients_getting_hot),
+  }
+  if hot_msg then
+    self.world.ui.adviser:say(hot_msg[math.random(1, #hot_msg)])
+    self.warmth_msg = true
+  end
+end
+
+function HeatingController:afterLoad(old,new)
+  if old < 125 then
+    self:resetBoilerBreakdownCounter()
+  end
+end
+
+function HeatingController:coldWarning(isPlayerHospital)
```

Comment:
The test for announcements and the argument are redundant. These methods should be private (prefix with _ or specify outside of the class) and should not take isPlayerHospital.

Reply:
They need isPlayerHospital otherwise adviser will show messages